test/06-device_log.ts

```diff
@@ -0,0 +1,191 @@
+import * as _ from 'lodash';
+import { expect } from './test-lib/chai';
+import * as fixtures from './test-lib/fixtures';
+import { supertest } from './test-lib/supertest';
+
+import { app } from '../init';
+
+describe('device log', () => {
+	const ctx: AnyObject = {};
+	before(async () => {
+		const fx = await fixtures.load('create-device-log');
+		ctx.loadedFixtures = fx;
+		ctx.user = fx.users.admin;
+		ctx.device = fx.devices.device1;
+		ctx.device2 = fx.devices.device2;
+
+		const res = await supertest(app, ctx.user)
+			.post(`/api-key/device/${ctx.device.id}/device-key`)
+			.expect(200);
+
+		expect(res.body).to.be.a('string');
+		ctx.device.apiKey = res.body;
+
+		const res2 = await supertest(app, ctx.user)
+			.post('/api-key/user/full')
+			.send({ name: 'name' })
+			.expect(200);
+
+		expect(res2.body).to.be.a('string');
+		ctx.user.apiKey = res2.body;
+	});
+	after(() => fixtures.clean(ctx.loadedFixtures));
+
+	const createLog = (extra = {}) => {
+		return {
+			isStdErr: true,
+			isSystem: true,
+			message: 'a log line',
+			timestamp: Date.now(),
+			...extra,
+		};
+	};
```

Comment:
Could you move this out of the `describe` altogether please, it'll make it clear it doesn't require any context from this test block

Reply:
moved